core/services/chainlink/application.go

```diff
@@ -249,6 +244,20 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 		return nil, fmt.Errorf("no evm chains found")
 	}
 
+	var (
+		pipelineORM    = pipeline.NewORM(db, globalLogger, cfg.Database(), cfg.JobPipeline().MaxSuccessfulRuns())
+		bridgeORM      = bridges.NewORM(db, globalLogger, cfg.Database())
+		mercuryORM     = mercury.NewORM(db, globalLogger, cfg.Database())
+		pipelineRunner = pipeline.NewRunner(pipelineORM, bridgeORM, cfg.JobPipeline(), cfg.WebServer(), legacyEVMChains, keyStore.Eth(), keyStore.VRF(), globalLogger, restrictedHTTPClient, unrestrictedHTTPClient)
+		jobORM         = job.NewORM(db, pipelineORM, bridgeORM, keyStore, globalLogger, cfg.Database())
+		txmORM         = txmgr.NewTxStore(db, globalLogger, cfg.Database())
+	)
+
+	srvcs = append(srvcs, eventBroadcaster, mailMon)
+	srvcs = append(srvcs, relayerChainInterops.Services()...)
+	promReporter := promreporter.NewPromReporter(txmORM, globalLogger)
```

Comment:
Since we don't want to use TxStore directly here, I think you'll need to use `legacyEVMChains` from above. That should contain [chain objects](https://github.com/smartcontractkit/chainlink/blob/54563c01b188289c4d8aa3381bd2f6905c402161/core/chains/evm/chain.go#L111-L125) that you can get the TXM from for each.